docs/resources/server/loottables/datagen.md

```diff
@@ -0,0 +1,200 @@
+# Loot Table Datagen
+
+[Loot tables][loottable] can be [datagenned][datagen] by subclassing `LootTableProvider` and providing a list of `LootTableSubProvider` in the constructor. Due to Mojang being Mojang, this requires some boilerplate to fully set up:
+
+```java
+public class MyLootTableProvider extends LootTableProvider {
+    // Get the PackOutput from GatherDataEvent.
+    public MyLootTableProvider(PackOutput output) {
+        super(output,
+                // A set of required table resource locations. This is used in validation (see below).
+                // It is generally not recommended for mods to validate, therefore we pass in an empty set.
+                Set.of(),
+                // A list of sub provider entries. See below for what values to use here.
+                List.of());
```

Comment:
Ok, I've been seeing this around. I would rather you represent something you're going to do later with `...` or something since people are likely just to see the empty list, copy and paste, and then go it didn't work.